WeakAuras/Types.lua

```diff
@@ -1635,15 +1635,34 @@ local function InitializeReputations()
   ---@type table<string, boolean>
   Private.reputations_headers = {}
 
+  -- Ensure all factions are shown by adjusting filters
+  local showLegacy = true
+  if not Private.ExecEnv.AreLegacyReputationsShown() then
+    showLegacy = false
+    C_Reputation.SetLegacyReputationsShown(true)
+  end
+  local sortType = 0
+  if Private.ExecEnv.GetReputationSortType() > 0 then
+    sortType = Private.ExecEnv.GetReputationSortType()
+    C_Reputation.SetReputationSortType(0)
+  end
+
+  -- Dynamic expansion of all collapsed headers
   local collapsed = {}
-  for i = 1, Private.ExecEnv.GetNumFactions() do
-    local factionData = Private.ExecEnv.GetFactionDataByIndex(i)
-    if factionData.isCollapsed then
-      collapsed[factionData.name] = true
+  local keepExpanding = true
+  while keepExpanding do
+    keepExpanding = false
+    for i = Private.ExecEnv.GetNumFactions(), 1, -1 do
```

Comment:
Can't you iterate forward and then don't need the outer loop?

Reply:
I've tried this:
```lua
  local collapsed = {}
  local index = 1
  local keepExpanding = true
  while keepExpanding do
    keepExpanding = false
    local factionData = Private.ExecEnv.GetFactionDataByIndex(index)
    if factionData.isHeader and factionData.isCollapsed then
      Private.ExecEnv.ExpandFactionHeader(index)
      collapsed[factionData.name] = true
    end
    if factionData.name then
      keepExpanding = true
    end
    index = index + 1
  end
  ```
  
  And although that collects all factions, it doesn't correctly restore collapsed headers later on.
  I think that is because of even more weird interactions in this API:
![nEpCp2b](https://github.com/WeakAuras/WeakAuras2/assets/6916881/f3528eb1-a444-4a29-bac9-bd93502eaf95)

Reply:
I guess not doing a reverse loop somehow breaks index

Reply:
I pushed something to your branch which iterates forward (and thus only once). I can't reproduce the issue you show in that gif.